### ISSUE TITLE
Improve favicon

### ISSUE THREAD

**Issue by kelson42** [openzim/gutenberg#165]
We have only a minimal favicon but we should have a full favicon support for all possible devices. See https://realfavicongenerator.net/favicon_checker?protocol=https&site=library.kiwix.org%2Fcontent%2Fgutenberg_fr_all%2FA%2FHome.html

**Comment by benoit74** [comment]
@eshellman, is it still appropriate to use this logo : https://commons.wikimedia.org/wiki/File:Project_Gutenberg_logo.svg converted in the various formats needed ?

@kelson42 @rgaudin could you point a scrapper already adding appropriate sizes / companion files (manifest, ...) so we could reuse the code?

**Comment by kelson42** [comment]
`openzim/phets` has it implemented but this is in Typescript. AFAIK, no Python scraper has implemented this so far.

**Comment by eshellman** [comment]
![pg-logo-144x144-original](https://user-images.githubusercontent.com/926513/214625424-03ed0655-ed7b-4d4c-a5d1-faf38784b0d3.png)
![pg-logo](https://user-images.githubusercontent.com/926513/214625445-97fb0564-18bb-4e96-b8ec-50b775898ac3.png)

**Comment by rgaudin** [comment]
@eshellman could you share the source of those images? Or at least higher resolution images. We'd need 512x512 for the large one and 48x48 for the simplified version (according to that website).

Is it only somewhere? Who's the author/license?

Finally, should the light color on the edges be removed (ie. have a rounded square on top of transparent background) ?

**Comment by eshellman** [comment]
I'll track that down. I have a transparent edge version of the png too.
![pg-logo-144x144-transparent](https://user-images.githubusercontent.com/926513/214872509-d0241cf4-105c-4de1-8abb-1aad20d709f7.png)

**Comment by kelson42** [comment]
@eshellman Thx, I guess we have now the data material to move forward!?

**Comment by benoit74** [comment]
Nope, as mentioned and detailed here https://github.com/gbnewby/gutenbergsite/issues/24#issuecomment-1683818588 we would benefit from an online file + a file in higher resolution.

**Comment by benoit74** [comment]
Good online resources to understand files / resolutions needed / recommended / expected:
- https://mobiforge.com/design-development/adding-favicons-in-a-multi-browser-multi-platform-world
- https://web.dev/icons-and-browser-colors/
- https://www.emergeinteractive.com/insights/detail/The-Essentials-of-FavIcons/
- https://evilmartians.com/chronicles/how-to-favicon-in-2021-six-files-that-fit-most-needs

Should we :
1. use RealFaviconGenerator API at https://realfavicongenerator.net/api/non_interactive_api everytime the scraper runs in order to generate proper data with explicit settings
2. generate HTML / files with RealFaviconGenerator once and embed it in the scraper code (meaning we will have to run it again at a regular interval)

I prefer option 2 because it reduces greatly our dependency on an online service which might disappear + allow us to easily switch to another one / tweak the result if needed + it allows more easily to upscale the source image used to generate other ones manually with AI or other tools manually before starting the process

**Comment by benoit74** [comment]
I upscaled the image with https://imgupscaler.com/ (randomly chosen, no strong experience on those tools) and it gave very decent result even if some zones are a bit blurry. Resulting image is 516x516 ... sic

![gutenberg_favicon_upscaled](https://github.com/openzim/gutenberg/assets/7102089/2da2ea0e-d208-4881-bd3d-2ba81d0ab786)

**Comment by mrcook** [comment]
By the looks of things this is an [Albion Hand Press](https://c8.alamy.com/comp/KBYHR9/eb1911-printing-payne-sons-albion-hand-press-KBYHR9.jpg):
![image](https://github.com/openzim/gutenberg/assets/1093101/26125586-7a39-414d-a810-da6395f668fc)

I'm sure the font would be easy to reproduce. Your favourite search engine would like produce an un-watermarked image.

**Comment by eshellman** [comment]
wow, that's pretty good!

**Comment by benoit74** [comment]
@siemsie can you help on this issue? We need a 48x48 nice-looking icon for the Gutenberg ZIM, I can brief you a bit if needed.

**Comment by siemsie** [comment]
Hi all. I don't know if I'm "solving the issue", but there is a cleaner version of the G on the red background which I have put in these 2 pngs.
Note that I attempted working with the original printing press and the P+G, it's ok at 512, but a mess at 48px as there is too much need for detail. For the purpose of clarity I suggest sticking to the G on red. Let me know! Thanks.

![Image](https://github.com/user-attachments/assets/316225ec-e728-4acc-bc2c-a5e3407dda0e)
![Image](https://github.com/user-attachments/assets/3745a2cc-0636-4f6b-8481-d7f934dc50d1)

**Comment by eshellman** [comment]
lots of versions here https://github.com/gutenbergtools/gutenbergsite/tree/master/gutenberg 

the block G logo persists in a few places, but PG prefers use of the current logo

**Comment by benoit74** [comment]
OK then I will just go with https://github.com/gutenbergtools/gutenbergsite/blob/master/gutenberg/pg-logo-144x144-transparent.png

144x144 is sufficient for now, and the fact that there is a border and a white zone outside the border is just your logo. Not our call.

**Comment by benoit74** [comment]
![Image](https://github.com/user-attachments/assets/fc304615-c8c8-4f75-af94-f29ed946ddb1)

**Comment by siemsie** [comment]
Here is the icon in 512px with letters and press.

![Image](https://github.com/user-attachments/assets/d472258c-a67f-4ed9-93cc-c49ea541d4ab)

**Comment by benoit74** [comment]
> We should probably better define the objective of that metadata. [spec](https://wiki.openzim.org/wiki/Metadata) says  to illustrate the zim file which is very vague and impossible to achieve. But distinguish from other ZIMs or recognizable are easier to achieve.

I can only second that point. And be clearer on what we expect from home HTML on https://realfavicongenerator.net (which is obviously distinct from the ZIM illustration)

**Comment by kelson42** [comment]
> > We should probably better define the objective of that metadata. [spec](https://wiki.openzim.org/wiki/Metadata) says  to illustrate the zim file which is very vague and impossible to achieve. But distinguish from other ZIMs or recognizable are easier to achieve.
> 
> I can only second that point. And be clearer on what we expect from home HTML on https://realfavicongenerator.net (which is obviously distinct from the ZIM illustration)

I suspect here a confusion between ZIM illustration metadata and HTML favicon. Here this is only about the HTML favicon AFAIK.

**Comment by benoit74** [comment]
@Lorrainnn would you like to work on this one? This is one of the two last issues pending before 2.2.0 release.

The goal is not to update the ZIM illustration (I've been mislead at some point) but to update the favicon(s) served on the various HTML pages of the ZIM. 

I'm not an expert on this topic, but would I develop this issue I would:
- upload illustration from https://github.com/openzim/gutenberg/issues/165#issuecomment-2775734747 to https://realfavicongenerator.net/
- use corresponding files / HTML code in all templates (probably only `base.html` has to be modified, but tbc)
  - you might want to modify this code to use new image at high res, but keep the red G icon at lower res since its rendering is definitely better ... no strong PoV on this

Beware about `favicon.png` which is a bit special since it is used as ZIM illustration + as icon in some stuff modified by JS code (didn't really understood yet what it is). We should probably keep this intact at this stage.

**Comment by Lorrainnn** [comment]
Sur.  I ll look into the issue. Hopefully to see 2.2.0 release soon! :)
